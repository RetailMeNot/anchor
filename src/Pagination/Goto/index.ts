export { Goto } from './Goto.components';
