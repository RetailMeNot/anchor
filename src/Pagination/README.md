# Pagination

The Pagination component is used to navigate between pages of results

## Notes

There are two ways to define the total number of pages. First, by setting the `totalPages` prop defines how many pages there are to tab between. Second, if you instead set the `totalResults` prop, then it'll calculate the number of total pages using the value of the `pageSize` prop.

Pagination can be "controlled" by defining the `current` prop to set the current page, rather than rely on internal state. This should be used in tandem with the `onChange` prop.
