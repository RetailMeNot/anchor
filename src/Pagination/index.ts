export { Pagination } from './Pagination.component';
