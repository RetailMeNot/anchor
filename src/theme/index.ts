export { default as colors } from './colors.theme';
export { default as fonts } from './fonts.theme';
export { default as sizes } from './sizes.theme';
export { default as variables } from './variables.theme';

export { NormalizeCSS, GlobalCSS } from './GlobalStyles';
