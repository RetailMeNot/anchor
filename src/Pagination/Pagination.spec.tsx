// VENDOR
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { ThemeProvider } from '@xstyled/styled-components';
import { shallow, mount } from 'enzyme';
// ANCHOR
import { RootTheme } from '../../src/theme';
import { Pagination } from './Pagination.component';

describe('Component: Pagination', () => {
    it('should be defined.', () => {
        const subject = (
            <ThemeProvider theme={RootTheme}>
                <Pagination totalPages={5} />
            </ThemeProvider>
        );
        const wrapper = mount(subject);
        const component = shallow(subject);

        expect(subject).toBeDefined();
        expect(wrapper).toBeDefined();
        expect(component).toBeDefined();
        const tree = renderer.create(subject).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should render with many pages.', () => {
        const subject = (
            <ThemeProvider theme={RootTheme}>
                <Pagination totalPages={100} />
            </ThemeProvider>
        );
        const tree = renderer.create(subject).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should render with current page in the middle.', () => {
        const subject = (
            <ThemeProvider theme={RootTheme}>
                <Pagination totalPages={20} current={10} />
            </ThemeProvider>
        );
        const tree = renderer.create(subject).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should render on the last page.', () => {
        const subject = (
            <ThemeProvider theme={RootTheme}>
                <Pagination totalPages={20} current={20} />
            </ThemeProvider>
        );
        const tree = renderer.create(subject).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
