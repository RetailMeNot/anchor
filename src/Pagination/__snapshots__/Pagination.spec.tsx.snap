// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Pagination should be defined. 1`] = `
.c3 {
  position: absolute;
  top: -1px;
  left: -1px;
  right: -1px;
  bottom: -1px;
  margin: -0.25rem -0.25rem;
}

.c2 {
  position: relative;
  border-radius: 4px;
  font-weight: 600;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  text-align: center;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none !important;
  margin: 0;
  -webkit-transition: 500ms ease background-color, 500ms ease border-color, 500ms ease color, 500ms ease fill;
  transition: 500ms ease background-color, 500ms ease border-color, 500ms ease color, 500ms ease fill;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: solid thin #D3D3D3;
  background-color: #D3D3D3;
  color: #808080;
  cursor: not-allowed;
  padding: 0;
  font-size: 0.875rem;
  height: 2.5rem;
  min-width: 2.5rem;
}

.c2 + .c1 {
  margin-left: 1rem;
}

.c5 {
  position: relative;
  border-radius: 4px;
  font-weight: 600;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  text-align: center;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none !important;
  margin: 0;
  -webkit-transition: 500ms ease background-color, 500ms ease border-color, 500ms ease color, 500ms ease fill;
  transition: 500ms ease background-color, 500ms ease border-color, 500ms ease color, 500ms ease fill;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: solid thin #0998D6;
  background-color: #0998D6;
  color: #FFFFFF;
  padding: 0;
  font-size: 0.875rem;
  height: 2.5rem;
  min-width: 2.5rem;
}

.c5 + .c1 {
  margin-left: 1rem;
}

.c5:hover,
.c5:focus,
.c5:active {
  background-color: #0074A6;
  border: solid thin #0074A6;
}

.c6 {
  position: relative;
  border-radius: 4px;
  font-weight: 600;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  text-align: center;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none !important;
  margin: 0;
  -webkit-transition: 500ms ease background-color, 500ms ease border-color, 500ms ease color, 500ms ease fill;
  transition: 500ms ease background-color, 500ms ease border-color, 500ms ease color, 500ms ease fill;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: solid thin transparent;
  background-color: transparent;
  color: #0074A6;
  padding: 0;
  font-size: 0.875rem;
  height: 2.5rem;
  min-width: 2.5rem;
}

.c6 + .c1 {
  margin-left: 1rem;
}

.c6:hover,
.c6:focus,
.c6:active {
  background: rgba(128,128,128,0.16);
  color: #0074A6;
}

.c4 {
  display: inline-block;
  height: 1.5rem;
  width: 1.5rem;
  line-height: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="anchor-pagination c0"
>
  <button
    className="anchor-button c1 c2"
    disabled={true}
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    <span
      className="anchor-icon chevron-left anchor-button-prefix c4"
      scale="lg"
    >
      <svg
        height={24}
        viewBox="0 0 16 16"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.522 8l2.518 2.953a.77.77 0 0 1 0 1.058l-.015.015a.721.721 0 0 1-1.052-.01L5.96 8.53a.767.767 0 0 1-.009-1.048l3.03-3.507a.719.719 0 0 1 1.043 0l.016.015c.28.293.28.765.01 1.047L7.522 8z"
          fill="currentColor"
          fillRule="nonzero"
        />
      </svg>
    </span>
  </button>
  <button
    className="anchor-button c1 c5"
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    1
  </button>
  <button
    className="anchor-button c1 c6"
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    2
  </button>
  <button
    className="anchor-button c1 c6"
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    3
  </button>
  <button
    className="anchor-button c1 c6"
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    4
  </button>
  <button
    className="anchor-button c1 c6"
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    5
  </button>
  <button
    className="anchor-button c1 c6"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    <span
      className="anchor-icon chevron-right anchor-button-prefix c4"
      scale="lg"
    >
      <svg
        height={24}
        viewBox="0 0 16 16"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.478 8L5.96 5.047a.77.77 0 0 1 .001-1.058l.014-.015a.721.721 0 0 1 1.052.01L10.04 7.47c.28.293.28.765.009 1.048l-3.03 3.507a.719.719 0 0 1-1.043 0l-.015-.015a.767.767 0 0 1-.01-1.047L8.477 8z"
          fill="currentColor"
          fillRule="nonzero"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`Component: Pagination should render on the last page. 1`] = `
.c3 {
  position: absolute;
  top: -1px;
  left: -1px;
  right: -1px;
  bottom: -1px;
  margin: -0.25rem -0.25rem;
}

.c7 {
  position: relative;
  border-radius: 4px;
  font-weight: 600;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  text-align: center;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none !important;
  margin: 0;
  -webkit-transition: 500ms ease background-color, 500ms ease border-color, 500ms ease color, 500ms ease fill;
  transition: 500ms ease background-color, 500ms ease border-color, 500ms ease color, 500ms ease fill;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: solid thin #D3D3D3;
  background-color: #D3D3D3;
  color: #808080;
  cursor: not-allowed;
  padding: 0;
  font-size: 0.875rem;
  height: 2.5rem;
  min-width: 2.5rem;
}

.c7 + .c1 {
  margin-left: 1rem;
}

.c6 {
  position: relative;
  border-radius: 4px;
  font-weight: 600;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  text-align: center;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none !important;
  margin: 0;
  -webkit-transition: 500ms ease background-color, 500ms ease border-color, 500ms ease color, 500ms ease fill;
  transition: 500ms ease background-color, 500ms ease border-color, 500ms ease color, 500ms ease fill;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: solid thin #0998D6;
  background-color: #0998D6;
  color: #FFFFFF;
  padding: 0;
  font-size: 0.875rem;
  height: 2.5rem;
  min-width: 2.5rem;
}

.c6 + .c1 {
  margin-left: 1rem;
}

.c6:hover,
.c6:focus,
.c6:active {
  background-color: #0074A6;
  border: solid thin #0074A6;
}

.c2 {
  position: relative;
  border-radius: 4px;
  font-weight: 600;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  text-align: center;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none !important;
  margin: 0;
  -webkit-transition: 500ms ease background-color, 500ms ease border-color, 500ms ease color, 500ms ease fill;
  transition: 500ms ease background-color, 500ms ease border-color, 500ms ease color, 500ms ease fill;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: solid thin transparent;
  background-color: transparent;
  color: #0074A6;
  padding: 0;
  font-size: 0.875rem;
  height: 2.5rem;
  min-width: 2.5rem;
}

.c2 + .c1 {
  margin-left: 1rem;
}

.c2:hover,
.c2:focus,
.c2:active {
  background: rgba(128,128,128,0.16);
  color: #0074A6;
}

.c4 {
  display: inline-block;
  height: 1.5rem;
  width: 1.5rem;
  line-height: 0;
}

.c5 {
  display: inline-block;
  height: 1rem;
  width: 1rem;
  line-height: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="anchor-pagination c0"
>
  <button
    className="anchor-button c1 c2"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    <span
      className="anchor-icon chevron-left anchor-button-prefix c4"
      scale="lg"
    >
      <svg
        height={24}
        viewBox="0 0 16 16"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.522 8l2.518 2.953a.77.77 0 0 1 0 1.058l-.015.015a.721.721 0 0 1-1.052-.01L5.96 8.53a.767.767 0 0 1-.009-1.048l3.03-3.507a.719.719 0 0 1 1.043 0l.016.015c.28.293.28.765.01 1.047L7.522 8z"
          fill="currentColor"
          fillRule="nonzero"
        />
      </svg>
    </span>
  </button>
  <button
    className="anchor-button c1 c2"
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    1
  </button>
  <button
    className="anchor-button c1 c2"
  >
    <div
      className="c3"
    />
    <span
      className="anchor-icon ellipses anchor-button-prefix c5"
      scale="md"
    >
      <svg
        height={16}
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <defs>
          <path
            d="M3.5 10a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zM8 10a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm4.5 0a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3z"
            id="ellipsis-a"
          />
        </defs>
        <g
          fill="none"
          fillRule="evenodd"
        >
          <path
            d="M0 0h16v16H0z"
          />
          <use
            fill="currentColor"
            xlinkHref="#ellipsis-a"
          />
        </g>
      </svg>
    </span>
  </button>
  <button
    className="anchor-button c1 c2"
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    16
  </button>
  <button
    className="anchor-button c1 c2"
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    17
  </button>
  <button
    className="anchor-button c1 c2"
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    18
  </button>
  <button
    className="anchor-button c1 c2"
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    19
  </button>
  <button
    className="anchor-button c1 c6"
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    20
  </button>
  <button
    className="anchor-button c1 c7"
    disabled={true}
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    <span
      className="anchor-icon chevron-right anchor-button-prefix c4"
      scale="lg"
    >
      <svg
        height={24}
        viewBox="0 0 16 16"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.478 8L5.96 5.047a.77.77 0 0 1 .001-1.058l.014-.015a.721.721 0 0 1 1.052.01L10.04 7.47c.28.293.28.765.009 1.048l-3.03 3.507a.719.719 0 0 1-1.043 0l-.015-.015a.767.767 0 0 1-.01-1.047L8.477 8z"
          fill="currentColor"
          fillRule="nonzero"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`Component: Pagination should render with current page in the middle. 1`] = `
.c3 {
  position: absolute;
  top: -1px;
  left: -1px;
  right: -1px;
  bottom: -1px;
  margin: -0.25rem -0.25rem;
}

.c7 + .c1 {
  margin-left: 1rem;
}

.c6 {
  position: relative;
  border-radius: 4px;
  font-weight: 600;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  text-align: center;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none !important;
  margin: 0;
  -webkit-transition: 500ms ease background-color, 500ms ease border-color, 500ms ease color, 500ms ease fill;
  transition: 500ms ease background-color, 500ms ease border-color, 500ms ease color, 500ms ease fill;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: solid thin #0998D6;
  background-color: #0998D6;
  color: #FFFFFF;
  padding: 0;
  font-size: 0.875rem;
  height: 2.5rem;
  min-width: 2.5rem;
}

.c6 + .c1 {
  margin-left: 1rem;
}

.c6:hover,
.c6:focus,
.c6:active {
  background-color: #0074A6;
  border: solid thin #0074A6;
}

.c2 {
  position: relative;
  border-radius: 4px;
  font-weight: 600;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  text-align: center;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none !important;
  margin: 0;
  -webkit-transition: 500ms ease background-color, 500ms ease border-color, 500ms ease color, 500ms ease fill;
  transition: 500ms ease background-color, 500ms ease border-color, 500ms ease color, 500ms ease fill;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: solid thin transparent;
  background-color: transparent;
  color: #0074A6;
  padding: 0;
  font-size: 0.875rem;
  height: 2.5rem;
  min-width: 2.5rem;
}

.c2 + .c1 {
  margin-left: 1rem;
}

.c2:hover,
.c2:focus,
.c2:active {
  background: rgba(128,128,128,0.16);
  color: #0074A6;
}

.c4 {
  display: inline-block;
  height: 1.5rem;
  width: 1.5rem;
  line-height: 0;
}

.c5 {
  display: inline-block;
  height: 1rem;
  width: 1rem;
  line-height: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="anchor-pagination c0"
>
  <button
    className="anchor-button c1 c2"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    <span
      className="anchor-icon chevron-left anchor-button-prefix c4"
      scale="lg"
    >
      <svg
        height={24}
        viewBox="0 0 16 16"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.522 8l2.518 2.953a.77.77 0 0 1 0 1.058l-.015.015a.721.721 0 0 1-1.052-.01L5.96 8.53a.767.767 0 0 1-.009-1.048l3.03-3.507a.719.719 0 0 1 1.043 0l.016.015c.28.293.28.765.01 1.047L7.522 8z"
          fill="currentColor"
          fillRule="nonzero"
        />
      </svg>
    </span>
  </button>
  <button
    className="anchor-button c1 c2"
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    1
  </button>
  <button
    className="anchor-button c1 c2"
  >
    <div
      className="c3"
    />
    <span
      className="anchor-icon ellipses anchor-button-prefix c5"
      scale="md"
    >
      <svg
        height={16}
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <defs>
          <path
            d="M3.5 10a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zM8 10a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm4.5 0a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3z"
            id="ellipsis-a"
          />
        </defs>
        <g
          fill="none"
          fillRule="evenodd"
        >
          <path
            d="M0 0h16v16H0z"
          />
          <use
            fill="currentColor"
            xlinkHref="#ellipsis-a"
          />
        </g>
      </svg>
    </span>
  </button>
  <button
    className="anchor-button c1 c2"
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    9
  </button>
  <button
    className="anchor-button c1 c6"
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    10
  </button>
  <button
    className="anchor-button c1 c2"
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    11
  </button>
  <button
    className="anchor-button c1 c2"
  >
    <div
      className="c3"
    />
    <span
      className="anchor-icon ellipses anchor-button-prefix c5"
      scale="md"
    >
      <svg
        height={16}
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <defs>
          <path
            d="M3.5 10a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zM8 10a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm4.5 0a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3z"
            id="ellipsis-a"
          />
        </defs>
        <g
          fill="none"
          fillRule="evenodd"
        >
          <path
            d="M0 0h16v16H0z"
          />
          <use
            fill="currentColor"
            xlinkHref="#ellipsis-a"
          />
        </g>
      </svg>
    </span>
  </button>
  <button
    className="anchor-button c1 c2"
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    20
  </button>
  <button
    className="anchor-button c1 c2"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    <span
      className="anchor-icon chevron-right anchor-button-prefix c4"
      scale="lg"
    >
      <svg
        height={24}
        viewBox="0 0 16 16"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.478 8L5.96 5.047a.77.77 0 0 1 .001-1.058l.014-.015a.721.721 0 0 1 1.052.01L10.04 7.47c.28.293.28.765.009 1.048l-3.03 3.507a.719.719 0 0 1-1.043 0l-.015-.015a.767.767 0 0 1-.01-1.047L8.477 8z"
          fill="currentColor"
          fillRule="nonzero"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`Component: Pagination should render with many pages. 1`] = `
.c3 {
  position: absolute;
  top: -1px;
  left: -1px;
  right: -1px;
  bottom: -1px;
  margin: -0.25rem -0.25rem;
}

.c2 {
  position: relative;
  border-radius: 4px;
  font-weight: 600;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  text-align: center;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none !important;
  margin: 0;
  -webkit-transition: 500ms ease background-color, 500ms ease border-color, 500ms ease color, 500ms ease fill;
  transition: 500ms ease background-color, 500ms ease border-color, 500ms ease color, 500ms ease fill;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: solid thin #D3D3D3;
  background-color: #D3D3D3;
  color: #808080;
  cursor: not-allowed;
  padding: 0;
  font-size: 0.875rem;
  height: 2.5rem;
  min-width: 2.5rem;
}

.c2 + .c1 {
  margin-left: 1rem;
}

.c5 {
  position: relative;
  border-radius: 4px;
  font-weight: 600;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  text-align: center;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none !important;
  margin: 0;
  -webkit-transition: 500ms ease background-color, 500ms ease border-color, 500ms ease color, 500ms ease fill;
  transition: 500ms ease background-color, 500ms ease border-color, 500ms ease color, 500ms ease fill;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: solid thin #0998D6;
  background-color: #0998D6;
  color: #FFFFFF;
  padding: 0;
  font-size: 0.875rem;
  height: 2.5rem;
  min-width: 2.5rem;
}

.c5 + .c1 {
  margin-left: 1rem;
}

.c5:hover,
.c5:focus,
.c5:active {
  background-color: #0074A6;
  border: solid thin #0074A6;
}

.c6 {
  position: relative;
  border-radius: 4px;
  font-weight: 600;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  text-align: center;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none !important;
  margin: 0;
  -webkit-transition: 500ms ease background-color, 500ms ease border-color, 500ms ease color, 500ms ease fill;
  transition: 500ms ease background-color, 500ms ease border-color, 500ms ease color, 500ms ease fill;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: solid thin transparent;
  background-color: transparent;
  color: #0074A6;
  padding: 0;
  font-size: 0.875rem;
  height: 2.5rem;
  min-width: 2.5rem;
}

.c6 + .c1 {
  margin-left: 1rem;
}

.c6:hover,
.c6:focus,
.c6:active {
  background: rgba(128,128,128,0.16);
  color: #0074A6;
}

.c4 {
  display: inline-block;
  height: 1.5rem;
  width: 1.5rem;
  line-height: 0;
}

.c7 {
  display: inline-block;
  height: 1rem;
  width: 1rem;
  line-height: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="anchor-pagination c0"
>
  <button
    className="anchor-button c1 c2"
    disabled={true}
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    <span
      className="anchor-icon chevron-left anchor-button-prefix c4"
      scale="lg"
    >
      <svg
        height={24}
        viewBox="0 0 16 16"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.522 8l2.518 2.953a.77.77 0 0 1 0 1.058l-.015.015a.721.721 0 0 1-1.052-.01L5.96 8.53a.767.767 0 0 1-.009-1.048l3.03-3.507a.719.719 0 0 1 1.043 0l.016.015c.28.293.28.765.01 1.047L7.522 8z"
          fill="currentColor"
          fillRule="nonzero"
        />
      </svg>
    </span>
  </button>
  <button
    className="anchor-button c1 c5"
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    1
  </button>
  <button
    className="anchor-button c1 c6"
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    2
  </button>
  <button
    className="anchor-button c1 c6"
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    3
  </button>
  <button
    className="anchor-button c1 c6"
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    4
  </button>
  <button
    className="anchor-button c1 c6"
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    5
  </button>
  <button
    className="anchor-button c1 c6"
  >
    <div
      className="c3"
    />
    <span
      className="anchor-icon ellipses anchor-button-prefix c7"
      scale="md"
    >
      <svg
        height={16}
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <defs>
          <path
            d="M3.5 10a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zM8 10a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm4.5 0a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3z"
            id="ellipsis-a"
          />
        </defs>
        <g
          fill="none"
          fillRule="evenodd"
        >
          <path
            d="M0 0h16v16H0z"
          />
          <use
            fill="currentColor"
            xlinkHref="#ellipsis-a"
          />
        </g>
      </svg>
    </span>
  </button>
  <button
    className="anchor-button c1 c6"
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    100
  </button>
  <button
    className="anchor-button c1 c6"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="c3"
    />
    <span
      className="anchor-icon chevron-right anchor-button-prefix c4"
      scale="lg"
    >
      <svg
        height={24}
        viewBox="0 0 16 16"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.478 8L5.96 5.047a.77.77 0 0 1 .001-1.058l.014-.015a.721.721 0 0 1 1.052.01L10.04 7.47c.28.293.28.765.009 1.048l-3.03 3.507a.719.719 0 0 1-1.043 0l-.015-.015a.767.767 0 0 1-.01-1.047L8.477 8z"
          fill="currentColor"
          fillRule="nonzero"
        />
      </svg>
    </span>
  </button>
</div>
`;
